src/nodes.rs

```diff
@@ -86,6 +86,108 @@ pub async fn create_nodes(
     nodes
 }
 
+pub async fn create_nodes_filtered(
```

Comment:
It could be good to add even one test to check (especially for future) if it's working as expected.
I've checked that using below test (feel free to just copy&paste it).

```suggestion
#[cfg(test)]
mod tests {
    use crate::config::GeoIPConfiguration;
    use super::*;

    #[tokio::test]
    async fn create_nodes_filtered_test() {
        let indices = vec![vec![1, 2], vec![0, 2, 3], vec![0, 1, 3], vec![1, 2]];
        let node_addrs = vec![
            SocketAddr::from(([127, 0, 0, 1], 1234)),
            SocketAddr::from(([127, 0, 0, 2], 1234)),
            SocketAddr::from(([127, 0, 0, 3], 1234)),
            SocketAddr::from(([127, 0, 0, 4], 1234)),
        ];
        let node_network_types = vec![
            NetworkType::Unknown,
            NetworkType::Zcash,
            NetworkType::Unknown,
            NetworkType::Zcash,
        ];

        let config = GeoIPConfiguration::default();
        let mut geo_cache = GeoIPCache::new(&config);
        geo_cache.configure_providers(&config);

        let nodes = create_nodes_filtered(
            NetworkType::Zcash,
            &indices,
            &node_addrs,
            &node_network_types,
            &geo_cache,
        )
        .await;

        assert_eq!(nodes.len(), 2);
        assert_eq!(nodes[0].connections, vec![1]);
        assert_eq!(nodes[1].connections, vec![0]);

    }
}
```